lib/expression.ts

```diff
@@ -20,22 +20,83 @@ export default class ExpressionTranspiler extends base.TranspilerBase {
           this.visit(binExpr.right);
           this.emit(')');
         } else {
-          this.visit(binExpr.left);
-          if (operatorKind === ts.SyntaxKind.InstanceOfKeyword) {
-            this.emit('is');
-            this.fc.visitTypeName(<ts.Identifier>binExpr.right);
-          } else if (operatorKind == ts.SyntaxKind.InKeyword) {
-            this.reportError(node, 'in operator is unsupported');
-          } else {
-            this.emit(ts.tokenToString(binExpr.operatorToken.kind));
-            this.visit(binExpr.right);
+          var tokenStr = ts.tokenToString(operatorKind);
+          switch (operatorKind) {
+            case ts.SyntaxKind.CaretEqualsToken:
+            case ts.SyntaxKind.BarEqualsToken:
+            case ts.SyntaxKind.AmpersandEqualsToken:
+            case ts.SyntaxKind.GreaterThanGreaterThanEqualsToken:
+            case ts.SyntaxKind.LessThanLessThanEqualsToken:
+              this.visit(binExpr.left);
+              this.emit('=');
+              this.emit('(');
+              this.visit(binExpr.left);
+              this.emit('as int)');
+              this.emit(tokenStr.substring(0, tokenStr.indexOf('=')));
+              if (binExpr.right.kind == ts.SyntaxKind.Identifier) {
```

Comment:
I think this is not quite correct (think of e.g. function calls or whatever), and not strictly needed. Just drop it, and always emit the case.